privval/file.go

```diff
@@ -128,16 +138,21 @@ func (lss *FilePVLastSignState) CheckHRS(height int64, round int32, step int8) (
 
 // Save persists the FilePvLastSignState to its filePath.
 func (lss *FilePVLastSignState) Save() {
-	outFile := lss.filePath
-	if outFile == "" {
-		panic("cannot save FilePVLastSignState: filePath not set")
+	if lss.f == nil {
+		panic("cannot save FilePVLastSignState.f is nil")
 	}
+
 	jsonBytes, err := tmjson.MarshalIndent(lss, "", "  ")
 	if err != nil {
 		panic(err)
 	}
-	err = tempfile.WriteFileAtomic(outFile, jsonBytes, 0600)
-	if err != nil {
+	if _, err := lss.f.Seek(0, io.SeekStart); err != nil {
```

Comment:
Seek is not usually threadsafe, can we guarantee that this is only happening from within a single thread?